packages/@expo/cli/src/run/remoteBuildCache.ts

```diff
@@ -168,3 +175,24 @@ function importFingerprintForDev(projectRoot: string): null | typeof import('@ex
     return null;
   }
 }
+
+function isDevClientBuild({
+  runOptions,
+  projectRoot,
+}: {
+  runOptions: AndroidRunOptions | IosRunOptions;
+  projectRoot: string;
+}) {
+  if (!hasDirectDevClientDependency(projectRoot)) {
+    return false;
+  }
+
+  if ('variant' in runOptions && runOptions.variant) {
+    return runOptions.variant === 'debug';
+  }
+  if ('configuration' in runOptions && runOptions.configuration) {
+    return runOptions.configuration === 'Debug';
+  }
```

Comment:
```suggestion
  if ('variant' in runOptions) {
    return runOptions.variant === 'debug';
  }
  if ('configuration' in runOptions) {
    return runOptions.configuration === 'Debug';
  }
```

looks like redundant check

Reply:
I know that it looks a bit weird, but `runOptions.configuration` can be `undefined`
<img width="345" alt="image" src="https://github.com/user-attachments/assets/31d3645f-c5f8-4266-8e91-d50b1bf59d5e" />

Reply:
updated